include/Model/Languages.php

```diff
@@ -884,45 +884,71 @@ protected function update_translations( $old_slug, $new_slug = '' ): void {
 					} else {
 						$dr['id'][] = (int) $tr[ $old_slug ];
 						$dr['tt'][] = (int) $term->term_taxonomy_id;
 					}
 					unset( $tr[ $old_slug ] );
 
 					if ( empty( $tr ) || 1 == count( $tr ) ) {
-						$dt['t'][] = (int) $term->term_id;
+						$dt['t'][]  = (int) $term->term_id;
 						$dt['tt'][] = (int) $term->term_taxonomy_id;
 					} else {
-						$ut['case'][] = $wpdb->prepare( 'WHEN %d THEN %s', $term->term_id, maybe_serialize( $tr ) );
-						$ut['in'][] = (int) $term->term_id;
+						$ut['case'][] = array( $term->term_id, maybe_serialize( $tr ) );
+						$ut['in'][]   = (int) $term->term_id;
 					}
 				}
 			}
 		}
 
-		// Delete relationships
+		// Delete relationships.
 		if ( ! empty( $dr ) ) {
 			$wpdb->query(
-				"DELETE FROM $wpdb->term_relationships
-				WHERE object_id IN ( " . implode( ',', $dr['id'] ) . ' ) ' . // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
-				'AND term_taxonomy_id IN ( ' . implode( ',', $dr['tt'] ) . ' )' // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+				$wpdb->prepare(
+					sprintf(
+						"DELETE FROM {$wpdb->term_relationships} WHERE object_id IN (%s) AND term_taxonomy_id IN (%s)",
+						implode( ',', array_fill( 0, count( $dr['id'] ), '%d' ) ),
+						implode( ',', array_fill( 0, count( $dr['tt'] ), '%d' ) )
+					),
+					array_merge( $dr['id'], $dr['tt'] )
+				)
 			);
 		}
 
-		// Delete terms
+		// Delete terms.
 		if ( ! empty( $dt ) ) {
-			$wpdb->query( "DELETE FROM $wpdb->terms WHERE term_id IN ( " . implode( ',', $dt['t'] ) . ' )' ); // PHPCS:ignore WordPress.DB.PreparedSQL.NotPrepared
-			$wpdb->query( "DELETE FROM $wpdb->term_taxonomy WHERE term_taxonomy_id IN ( " . implode( ',', $dt['tt'] ) . ' )' ); // PHPCS:ignore WordPress.DB.PreparedSQL.NotPrepared
+			$wpdb->query(
+				$wpdb->prepare(
+					sprintf(
+						"DELETE FROM {$wpdb->terms} WHERE term_id IN (%s)",
+						implode( ',', array_fill( 0, count( $dt['t'] ), '%d' ) )
+					),
+					$dt['t']
+				)
+			);
+			$wpdb->query(
+				$wpdb->prepare(
+					sprintf(
+						"DELETE FROM {$wpdb->term_taxonomy} WHERE term_taxonomy_id IN (%s)",
+						implode( ',', array_fill( 0, count( $dt['tt'] ), '%d' ) )
+					),
+					$dt['tt']
+				)
+			);
 		}
 
-		// Update terms
+		// Update terms.
 		if ( ! empty( $ut ) ) {
 			$wpdb->query(
-				"UPDATE $wpdb->term_taxonomy
-				SET description = ( CASE term_id " . implode( ' ', $ut['case'] ) . ' END ) ' . // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
-				'WHERE term_id IN ( ' . implode( ',', $ut['in'] ) . ' )' // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+				$wpdb->prepare(
+					sprintf(
+						"UPDATE {$wpdb->term_taxonomy} SET description = ( CASE term_id %s END ) WHERE term_id IN (%s)",
+						implode( ' ', array_fill( 0, count( $ut['case'] ), 'WHEN %d THEN %s' ) ),
+						implode( ',', array_fill( 0, count( $ut['in'] ), '%d' ) )
+					),
+					array_merge( array_merge( ...$ut['case'] ), $ut['in'] )
```

Comment:
Is this the same result you'd expect? I'ma asking, before we lose readability here and I couldn't wrap my head around what we want to achieve here 😅 
```suggestion
					array_unique( ...$ut['case'] )+ $ut['in']
```

Reply:
We are passing parameters to `sprintf()` which must be ordered correctly to match the placeholders.
Let's say we have 2 terms to update, then the arrays could look like this:
```
$ut['case'] = array
	array( 32, 'serialized value' ),
	array( 45, 'other serialized value' ),
);

$ut['in'] = array( 32, 45 );
```
We want to obtain a flat array of parameters like this:
```
array( 32, 'serialized value', 45, 'other serialized value', 32, 45 )
```
So as you see your proposal is not correct as it will remove some items.